lib/playlistGenerator/ConcatSource.js

```diff
@@ -427,23 +423,24 @@ ConcatSource.prototype.checkDiscontinuity = function (fileInfo,track) {
             that.ptsStats.addSample(correlation);
 
             // reassign firstDTS in accordance with pts diff to avoid noise
-            if (that.inner.refPTS > 0) {
+            if (that.playlist.ptsReference.absolute > 0) {
                 // prevent wrap issues
-                var encDist = addWithOverflow(track.firstEncoderDTS-that.inner.refEncoderDTS,track.wrapEncoderDTS);
-                while( encDist >= track.wrapEncoderDTS / 2){
-                    that.logger.debug("firstDTS - that.inner.refPTS >= track.wrapEncoderDTS / 2. that.inner.refPTS=%j that.inner.refEncoderDTS=%j",
-                        that.inner.refPTS, that.inner.refEncoderDTS);
-                    that.inner.refPTS += track.wrapEncoderDTS / 2;
-                    that.inner.refEncoderDTS = addWithOverflow(that.inner.refEncoderDTS + track.wrapEncoderDTS / 2,track.wrapEncoderDTS);
-                    encDist = addWithOverflow(track.firstEncoderDTS-that.inner.refEncoderDTS,track.wrapEncoderDTS);
+                var encDist = dtsDistance(track.firstEncoderDTS,that.playlist.ptsReference.pts,track.wrapEncoderDTS),
+                    diffThreshold = track.wrapEncoderDTS / 4;
+                while( encDist >= diffThreshold){
+                    that.logger.debug("firstDTS - that.playlist.ptsReference.absolute >= threshold. that.playlist.ptsReference.absolute=%j that.playlist.ptsReference.pts=%j threshold=%j",
+                        that.playlist.ptsReference.absolute, that.playlist.ptsReference.pts,diffThreshold);
+                    that.playlist.ptsReference.absolute += diffThreshold;
+                    that.playlist.ptsReference.pts = addWithOverflow(that.playlist.ptsReference.pts + diffThreshold,track.wrapEncoderDTS);
+                    encDist = dtsDistance(track.firstEncoderDTS,that.playlist.ptsReference.pts,track.wrapEncoderDTS);
                 }
-                firstDTS = that.inner.refPTS + dtsDistance(track.firstEncoderDTS, that.inner.refEncoderDTS,track.wrapEncoderDTS);
+                firstDTS = that.playlist.ptsReference.absolute + encDist;
             }
             // Fill drift information for diagnostics purposes
-            diagDrift['deltaPts'] = track.firstDTS - that.inner.refPTS;
-            diagDrift['deltaClock'] = track.firstEncoderDTS - that.inner.refEncoderDTS;
-            diagDrift['refPts'] = that.inner.refPTS;
-            diagDrift['refEncoderDts'] = that.inner.refEncoderDTS;
+            diagDrift['deltaPts'] = track.firstDTS - that.playlist.ptsReference.absolute;
```

Comment:
please fix that instead of throwing the diagnosticsInfo  it will be exposed when calling toJson of the playlist generator